ledgeri.py

```diff
@@ -102,35 +109,52 @@ def sign_tx(self, tx):
             seq.reverse()
             seq_hex = ''.join('{:02x}'.format(x) for x in seq)
 
-            # We will not attempt to sign non-witness inputs but
-            # need information for pre-processing
             if psbt_in.non_witness_utxo:
                 segwit_inputs.append({"value":txin.prevout.serialize()+struct.pack("<Q", psbt_in.non_witness_utxo.vout[txin.prevout.n].nValue), "witness":True, "sequence":seq_hex})
-                continue
+                # We only need legacy inputs in the case where all inputs are legacy, we check
+                # later
+                ledger_prevtx = bitcoinTransaction(psbt_in.non_witness_utxo.serialize())
+                legacy_inputs.append(self.app.getTrustedInput(ledger_prevtx, txin.prevout.n))
+                legacy_inputs[-1]["sequence"] = seq_hex
+                has_legacy = True
             else:
                 segwit_inputs.append({"value":txin.prevout.serialize()+struct.pack("<Q", psbt_in.witness_utxo.nValue), "witness":True, "sequence":seq_hex})
+                has_segwit = True
 
             pubkeys = []
             signature_attempts = []
 
             scriptCode = b""
             witness_program = b""
-            if psbt_in.witness_utxo.is_p2sh():
-                # Look up redeemscript
+            if psbt_in.witness_utxo is not None and psbt_in.witness_utxo.is_p2sh():
                 redeemscript = psbt_in.redeem_script
                 witness_program += redeemscript
-            else:
+            elif psbt_in.witness_utxo is not None and psbt_in.non_witness_utxo.is_p2sh():
```

Comment:
This should be `psbt_in.non_witness_utxo`. Also, `non_witness_utxo` does not have a `is_p2sh()`